### PR TITLE
Fix small issues in docs/src/dev/extensions.rst

### PR DESCRIPTION
**Pull request by wangkuiyi**
## Proposed changes

I am learning how to add an MLX extension to apply the bitmask generated by XGrammar to the vocabulary logits generated by the LLM during mlx_lm.generate. I noticed some small issues (mostly preliminary English grammar issues) in extensions.rst. This pull request fixes them. 

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Review by angeloskath — APPROVED** [review]
Thanks!
